src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -457,6 +459,11 @@ where
             )
             .await?;
 
+        let _defer_func = scopeguard::guard(internal_table_ids, |internal_table_ids| {
```

Comment:
Why using `scopeguard::guard` here?

Reply:
while the catalog_manager write fail, shall we need to reove the processing_table ?

Reply:
Could you explain why remove_processing_table here? I'm missing of some context. Thanks.

Reply:
> Could you explain why remove_processing_table here? I'm missing of some context. Thanks.

IMO, 2 case ,we need to remove_processing_table
1. fail in creation, mv create fail, but some info remain in processing_table. when snapshot happend, subscriber may retain some unuseful catalog
2. after store catalog to meta store, we can gather the table_catalog from meta store. To avoid duplicating, we need to remove_processing_table